### PR TITLE
add time series queries to synthetix-data

### PR DESCRIPTION
**Pull request by dvd-schwrtz**

### CODE REVIEW

File: bin.js [hunk 1]
```diff
@@ -35,6 +35,14 @@ program.command('exchanges.total').action(async () => {
 	exchanges.total().then(console.log);
 });
 
+program
+	.command('exchanges.aggregate')
+	.option('-e, --entity <value>', 'The type of total - dailyTotals, fifteenMinuteTotals', 'dailyTotals')
```
Comment (jjgonecrypto) [inline bin.js]: perhaps just `1d` or `15m` as the granularity and you map that to the entity name?

**Review by jjgonecrypto — CHANGES_REQUESTED** [review]
See comment

File: bin.js (hunk 1 above)
Comment (dvd-schwrtz) [reply bin.js]: ok cool 👍

File: bin.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ program.command('exchanges.total').action(async () => {
 
 program
 	.command('exchanges.aggregate')
-	.option('-e, --entity <value>', 'The type of total - dailyTotals, fifteenMinuteTotals', 'dailyTotals')
+	.option('-e, --entity <value>', 'The type of total - 1d, 15m', '1d')
```
Comment (jjgonecrypto) [inline bin.js]: Shouldn't we call it something other than `entity`?

Comment (jjgonecrypto) [reply bin.js]: Granularity perhaps?

File: index.js [hunk 3]
```diff
@@ -223,12 +223,21 @@ module.exports = {
 		/**
 		 * Get the aggregate exchange totals based on time periods.
 		 */
-		aggregate({ entity = 'dailyTotals', max = 30 } = {}) {
+		aggregate({ entity = '1d', max = 30 } = {}) {
```
Comment (jjgonecrypto) [inline index.js]: and here as well - you can map granularity to a entity name

File: index.js [hunk 4]
```diff
@@ -220,6 +220,41 @@ module.exports = {
 				}))
 				.catch(err => console.error(err));
 		},
+		/**
+		 * Get the aggregate exchange totals based on time periods.
+		 */
+		aggregate({ timeSeries = '1d', max = 30 } = {}) {
+			let formattedEntity;
+			if (timeSeries === '1d') {
```
Comment (jjgonecrypto) [inline index.js]: [np] you can use an object for that `entityMap = { '1d': 'dailyTotals', '15m': 'fifteenMinuteTotals' }

**Review by jjgonecrypto — APPROVED** [review]

**Pull request merged**